File: python/n278_FirstBad.py
# https://leetcode-cn.com/problems/first-bad-version/
# The isBadVersion API is already defined for you.
# @param version, an integer
# @return a bool
# def isBadVersion(version):

class Solution:
    def firstBadVersion(self, n):
        """
        :type n: int
        :rtype: int
        """
        left = 1
        right = n
        while left < right:
            if isBadVersion((left + right) // 2) == True:
                right = (left + right) // 2
            else:
                left = (left + right) // 2 + 1
        return left 
        

if __name__ == "__main__":
    pass
        

File: cpp/n278_FirstBad.cpp
// https://leetcode-cn.com/problems/first-bad-version/
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>
using namespace std;
// Forward declaration of isBadVersion API.
bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
        int left = 1;
        int right = n;
        while (left < right){
            int mid = left/2 + right/2;
            if (isBadVersion(mid) == true) right = mid;
            else left = mid + 1;
        }
        return left;
    }
};

int main(){
    return 0;
}